scripts/stop_rules.py

```
@@ -0,0 +1,848 @@
+
+
+def fit_scaling(d, y, through_origin=True):
+    """Fit n* ~ a + b*d^p (least squares over a p grid). Returns (a,b,p,rmse).
+
+    ``through_origin`` (default) forces a=0: a molecule with no dihedrals needs no
+    search, so a positive-or-zero intercept is the physical constraint, and a free
+    fit can return an unphysical negative intercept (usually an artifact of an
+    under-populated/outlier d-bin -- filter those out before fitting). Forcing the
+    origin is typically within a few percent RMSE of the free fit on clean data.
+    """
+    d = np.asarray(d, float); y = np.asarray(y, float)
```

Ruff (0.15.18)

[error] Multiple statements on one line (semicolon)

(E702)